VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/participant-self-test/participant-self-test.component.ts

```diff
@@ -38,4 +41,22 @@ export class ParticipantSelfTestComponent extends BaseSelfTestComponent {
         this.logger.debug('restarting participant self-test');
         this.selfTestComponent.replayVideo();
     }
+
+    @HostListener('window:beforeunload', ['$event'])
```

Comment:
Could you add tests for this new set of code, including negative scenarios

Reply:
done